fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -144,54 +135,24 @@ public LuceneIndexMaintainer(@Nonnull final IndexMaintainerState state, @Nonnull
                 throw new RecordCoreArgumentException("Auto complete does not support scanning with continuation")
                         .addLogInfo(LogMessageKeys.INDEX_NAME, state.index.getName());
             }
-            return new LuceneAutoCompleteResultCursor(getSuggester(Tuple.fromStream(range.getLow().stream().skip(1))),
-                    range.getLow().getString(0), executor, scanProperties, state, highlightForAutoCompleteIfEnabled);
+            LuceneScanAutoComplete scanAutoComplete = (LuceneScanAutoComplete)scanBounds;
+            return new LuceneAutoCompleteResultCursor(getSuggester(scanAutoComplete.getGroupKey()), scanAutoComplete.getKeyToComplete(),
+                    executor, scanProperties, state, scanAutoComplete.getGroupKey(), highlightForAutoCompleteIfEnabled);
         }
 
-        String[] fieldNames = indexTextFields(state.index, state.store.getRecordMetaData());
-        if (scanType.equals(IndexScanType.BY_LUCENE_SPELLCHECK)) {
+        if (scanType.equals(LuceneScanTypes.BY_LUCENE_SPELL_CHECK)) {
             if (continuation != null) {
                 throw new RecordCoreArgumentException("Spellcheck does not currently support continuation scanning");
             }
-            return new LuceneSpellcheckRecordCursor(range.getLow().getString(0), executor,
-                    scanProperties, state, Tuple.fromStream(range.getLow().stream().skip(1)), fieldNames);
+            LuceneScanSpellCheck scanSpellcheck = (LuceneScanSpellCheck)scanBounds;
+            return new LuceneSpellCheckRecordCursor(scanSpellcheck.getFields(), scanSpellcheck.getWord(),
+                    executor, scanProperties, state, scanSpellcheck.getGroupKey());
```

Comment:
I see that we removed the field names from the arguments. Those are different than the default normalized field names provided by the component. I forget exactly how but I remember some issues with Liverpool when using normalized names and the nesting that we do etc.

Reply:
The field names are still there. They just moved to the front, because the various cursors had needlessly drifted apart in their signature. And they come in as a result of parameter binding, not by doing metadata operations here or parsing inside the cursor.